Fix storage-node linter issues

shamil-gadelshin opened the pull request:
Changes:
- introduced linter rules list 
- added prettier to the eslint
- fix numerous warnings.

shamil-gadelshin committed:
Prepare lint configs

- change linter rules set to Atlas
- add prettier
- add prettier rules

shamil-gadelshin committed:
Autofix linter issues

shamil-gadelshin committed:
storage-node: change linting rules

shamil-gadelshin committed:
storage-node: Refactor ranges.js

shamil-gadelshin committed:
storage-node: fix linter errors

shamil-gadelshin committed:
storage-node: fix linter warnings

shamil-gadelshin committed:
storage-node: Fix linter warnings.

shamil-gadelshin committed:
storage-node: Fix linter warnings.

shamil-gadelshin committed:
storage-node: Fix linter warnings.

shamil-gadelshin committed:
storage-node: Fix linter warnings.

shamil-gadelshin committed:
storage-node: Fix linter warnings.

shamil-gadelshin committed:
storage-node: Modify prettier config.

shamil-gadelshin committed:
storage-node: Fix files format.

shamil-gadelshin committed:
storage-node: Fix linter errors.

mnaamani commented on storage-node/.eslintrc.js:
```diff
@@ -1,290 +1,41 @@
-    }
+    extends: [
+        "esnext",
+        "esnext/style-guide",
```
With introduction of #891 We shouldn't add a unique code style for this project.
Also in general you will have a hard time if you add style rules, if they don't match with your prettier config, its best to drop all style rules, see the approach I took in #891, so all style comes purely from prettier.

I would recommend incorporating the above PR and dropping this `extends` array.
You will then need a few extra rule overrides during transition to TS:
```
    '@typescript-eslint/no-var-requires': 'warning',
    '@typescript-eslint/camelcase': 'warning',
```
You will then find a few more linter errors, most are easy to fix.

mnaamani replied:
There is one error that didn't make a lot of sense, it just said delete the `.` in:

```javascript
module.exports = function (x,y) {

}
```
The problem is that its an anonymous function, giving it a name seems to make linter happy.

mnaamani commented on storage-node/.eslintrc.js:
```diff
@@ -1,290 +1,41 @@
+			"rules": {
+				// Disabling Rules because of used chai lib:
+				// https://stackoverflow.com/questions/45079454/no-unused-expressions-in-mocha-chai-unit-test-using-standardjs
+				"no-unused-expressions": "off",
```
The dirty-chai approach seemed neater, at least reducing config in the eslint config file, although requiring some code changes.

mnaamani commented on storage-node/.prettierrc:
```diff
@@ -0,0 +1,8 @@
+	"arrowParens": "avoid",
+	"useTabs": false,
+	"tabWidth": 2
+}
```
we should drop project specific prettier configs in favor of global one which will come in #891 
If its premature to merge it now, perhaps you can copy the same settings.

mnaamani commented on storage-node/.eslintrc.js:
```diff
@@ -1,290 +1,41 @@
+		"import/no-commonjs": "off", // remove after converting to TS.
+		// Disabling Rules because of monorepo environment:
+		// https://github.com/benmosher/eslint-plugin-import/issues/1174
+		"import/no-extraneous-dependencies": "off",
```
I was interested to see this. I turned the rule to error but when I run it i get:

```
 | ackages/runtime-api/4m/Users/mokhtar/joystream/joystream/storage-node/packages/runtime-api/assets.js
 |   1:1  error  Definition for rule 'import/no-extraneous-dependencies' was not found  import/no-extraneous-dependencies
 | 
```

Do we need to add `eslint-config-import` package to dev dependencies ?

mnaamani commented:
last point, I had to add `eslint-plugin-prettier` package to dev dependencies in root package.json as I was getting error that the plugin was not found when running the linter. Not sure why.

shamil-gadelshin committed:
storage-node: change configuration

- update root package.json: add ‘eslint-plugin-prettier’
- change the lint command in the storage-node
- add new rule ‘no-console’ to the lint config

shamil-gadelshin committed:
storage-node: Fix linter warnings.

mnaamani commented:
Flawless victory, not even a single warning ;)

Other than seeing this when running lint:
`The babel/generator-star-spacing rule is deprecated. Please use the built in generator-star-spacing rule instead.`

LGTM
